pd-client/rpc_worker.go

```diff
@@ -116,23 +116,28 @@ func (w *rpcWorker) stop(err error) {
 func (w *rpcWorker) work() {
 	defer w.wg.Done()
 
-RECONNECT:
-	log.Infof("[pd] connect to pd server %v", w.addr)
-	conn, err := rpcConnect(w.addr)
-	if err != nil {
-		log.Warnf("[pd] failed connect pd server: %v, will retry later", err)
-
-		select {
-		case <-time.After(netIOTimeout):
-			goto RECONNECT
-		case <-w.quit:
-			return
+	// Add default schema "http://" to addrs.
+	urls := make([]string, 0, len(w.addrs))
+	for _, addr := range w.addrs {
+		if strings.Contains(addr, "://") {
+			urls = append(urls, addr)
+		} else {
+			urls = append(urls, "http://"+addr)
 		}
 	}
 
-	reader := bufio.NewReaderSize(deadline.NewDeadlineReader(conn, netIOTimeout), readBufferSize)
-	writer := bufio.NewWriterSize(deadline.NewDeadlineWriter(conn, netIOTimeout), writeBufferSize)
-	readwriter := bufio.NewReadWriter(reader, writer)
+RECONNECT:
+	conn, isLeader := w.connectUrls(urls)
+	if conn == nil {
+		return
+	}
+	readwriter := newReadWriter(conn)
+	log.Infof("[pd] connected to %v", conn.RemoteAddr())
+
+	// We can't connect to the leader now, try to reconnect later.
+	if !isLeader {
+		time.AfterFunc(reconnectPDTimeout, func() { conn.Close() })
```

Comment:
The document says that "Multiple goroutines may invoke methods on a Conn simultaneously".

I actually create another goroutine to retry to connect to the leader in the first implementation, but that seems a bit complicated. Maybe I should try to simplify that.